node/src/accountant/scanners/mod.rs

```diff
@@ -1349,11 +1391,139 @@ mod tests {
         ));
     }
 
+    #[test]
+    fn keeping_entries_consistent_and_aligned_is_so_important() {
+        let wallet_1 = make_wallet("abc");
+        let hash_1 = make_tx_hash(123);
+        let wallet_2 = make_wallet("def");
+        let hash_2 = make_tx_hash(345);
+        let wallet_3 = make_wallet("ghi");
+        let hash_3 = make_tx_hash(546);
+        let wallet_4 = make_wallet("jkl");
+        let hash_4 = make_tx_hash(678);
+        let pending_payables_owned = vec![
+            PendingPayable::new(wallet_1.clone(), hash_1),
+            PendingPayable::new(wallet_2.clone(), hash_2),
+            PendingPayable::new(wallet_3.clone(), hash_3),
+            PendingPayable::new(wallet_4.clone(), hash_4),
+        ];
+        let pending_payables_ref = pending_payables_owned
+            .iter()
+            .collect::<Vec<&PendingPayable>>();
+        let pending_payable_dao = PendingPayableDaoMock::new().fingerprints_rowids_result(vec![
+            (Some(4), hash_4),
+            (Some(1), hash_1),
+            (Some(3), hash_3),
+            (Some(2), hash_2),
+        ]);
+        let subject = PayableScannerBuilder::new()
+            .pending_payable_dao(pending_payable_dao)
+            .build();
+
+        let (existent, nonexistent) = subject
+            .separate_id_triples_by_existent_and_nonexistent_fingerprints(&pending_payables_ref);
```

Comment:
Wouldn't the test be stronger if we let one fingerprint with the nonexistent fingerprint. Just a suggestion. 🙏

Reply:
Ok

Reply:
Ok, I'd like if it was simpler, but I understand that it was needed to be this way. It makes me concerned if it'll cause bugs in the future. For the time being, I'll limit this concern to myself. 😥